### PR TITLE
todos

### PR DESCRIPTION
**Pull request by donizer**
[DEMO LINK](https://donizer.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,14 +26,19 @@ export const App: React.FC = () => {
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                visibleItems.length === 0
```
Comment (maxim2310) [inline src/App.tsx:30]:
```suggestion
                !visibleItems.length
```

File: src/components/TodoList/TodoList.tsx [hunk 2]
```diff
@@ -1,100 +1,43 @@
...
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+      <tbody>
+        {
+          visibleItems && getFilteredItemsByCondition(visibleItems)
+            .filter(item => item.title.toLowerCase()
+              .includes(query.toLowerCase()))
```
Comment (maxim2310) [inline src/components/TodoList/TodoList.tsx:35]: move it to separate func

File: src/components/TodoItem/TodoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  } = React.useContext(todoContext) as DefaultValueType;
+
+  function handleClick() {
+    const user = getUser(todo.userId);
+
+    // console.log(user, todo);
+
+    // user.then((person) => {
+    //   console.log(person.name);
+    // })
```
Comment (maxim2310) [inline src/components/TodoItem/TodoItem.tsx:28]: remove it

File: src/components/TodoModal/TodoModal.tsx [hunk 4]
```diff
@@ -1,49 +1,90 @@
...
+
+    setCurrentItem({ ...currentItem, user: currUser });
+  }
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const callback = React.useCallback(resolveUser, []);
+
+  React.useEffect(() => {
+    callback();
+  }, [callback]);
```
Comment (maxim2310) [inline src/components/TodoModal/TodoModal.tsx:26]: why do you need it separately

File: src/components/TodoModal/TodoModal.tsx [hunk 5]
```diff
@@ -1,49 +1,90 @@
...
+          ? (
+            <Loader />
+          ) : (
+            <div className="modal-card">
+              <header className="modal-card-head">
+                <div
+                  className="modal-card-title has-text-weight-medium"
+                  data-cy="modal-header"
+                >
+                  {`Todo #${currentItem.todo.id}`}
```
Comment (maxim2310) [inline src/components/TodoModal/TodoModal.tsx:43]: Would be great to use destructuring for better readability

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Good work, but lets make some improvement

**Review by volodymyr-soltys97 — APPROVED** [review]

**Pull request closed without merging**